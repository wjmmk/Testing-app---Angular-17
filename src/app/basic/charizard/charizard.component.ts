import { Component, OnInit } from '@angular/core';
import { CharizardService } from './charizard.service';
import { Pokemon } from '../interfaces';
import { CommonModule } from '@angular/common';
import { HttpClient } from '@angular/common/http';

@Component({
  selector: 'app-charizard',
  standalone: true,
  imports: [CommonModule],
  templateUrl: './charizard.component.html',
  styleUrl: './charizard.component.scss'
})
export class CharizardComponent implements OnInit {
   public charizard?: Pokemon;

   constructor( private charizardService: CharizardService) {}

   ngOnInit(): void {
       this.charizardService.getPokemon(6)
         .subscribe( async (pokemon) => {
          this.charizard = pokemon
          console.log(pokemon)
         })
   }
}
