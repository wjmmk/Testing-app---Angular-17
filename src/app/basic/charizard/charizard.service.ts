import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Pokemon } from '../interfaces/pokemon';

@Injectable({
  providedIn: 'root'
})
export class CharizardService {

  constructor( private http: HttpClient ) { }

   getPokemon( id: number ): Observable<Pokemon> {
    return this.http.get<Pokemon>(`https://pokeapi.co/api/v2/pokemon/${id}`)
  }
}
