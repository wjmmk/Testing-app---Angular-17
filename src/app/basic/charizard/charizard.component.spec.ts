import { ComponentFixture, TestBed } from '@angular/core/testing';
import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing'
import { CharizardComponent } from './charizard.component';
import { CharizardService } from './charizard.service';


describe('CharizardComponent', () => {
  let component: CharizardComponent;
  let fixture: ComponentFixture<CharizardComponent>;
  let compiled: HTMLElement;
  let service: CharizardService;
  let httpMock: HttpTestingController;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [CharizardComponent, HttpClientTestingModule]
    })
    .compileComponents();

    fixture = TestBed.createComponent(CharizardComponent);
    component = fixture.componentInstance;
    service = TestBed.inject(CharizardService)
    httpMock = TestBed.inject(HttpTestingController)

    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  test('should create', () => {
    expect(component).toBeTruthy();
  });

  test('Should make a match with the SnapShop', () => {
    expect(compiled.innerHTML).toMatchSnapshot();
  })

  test('Should show a loading at start', () => {
    const h2 = compiled.querySelector('h2');
    expect(h2?.textContent).toContain('Loading...')
  })

  test('Should load to Charizard immediately', () => {
    const dummyPokemon = {
      name: 'charizardo!!',
      sprites: {
        front_default: 'https://charizard.com/sprite.png'
      }
    }

    const request = httpMock.expectOne('https://pokeapi.co/api/v2/pokemon/6')
    expect(request.request.method).toBe('GET')
    request.flush(dummyPokemon)
    fixture.detectChanges();
    //console.log(compiled.innerHTML)

    const h3 = compiled.querySelector('h3');
    const img = compiled.querySelector('img');

    expect(h3?.textContent?.toLocaleLowerCase()).toContain(dummyPokemon.name.toLocaleLowerCase())
    expect(img?.src).toBe(dummyPokemon.sprites.front_default)
    expect(img?.alt).toBe(dummyPokemon.name)
  })
});
