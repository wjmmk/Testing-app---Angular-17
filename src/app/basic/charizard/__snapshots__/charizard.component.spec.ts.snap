// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharizardComponent Should make a match with the SnapShop 1`] = `
"<h1>Pok√©mon</h1><div><h2>Loading...</h2></div><!--bindings={
  "ng-reflect-ng-if-else": "[object Object]"
}--><!--container-->"
`;
